src/Configuration.ts

```diff
@@ -12,91 +13,97 @@ import { stPassword_type_ppt } from "../generated/PagamentiTelematiciPspNodoserv
 
 const localhost = "http://localhost";
 
-export const CONFIG = {
-  APPINSIGHTS_INSTRUMENTATIONKEY:
-    process.env.APPINSIGHTS_INSTRUMENTATIONKEY || "key",
-  APPINSIGHTS_DISABLE: process.env.APPINSIGHTS_DISABLE || "false",
-  APPINSIGHTS_SAMPLING_PERCENTAGE:
-    process.env.APPINSIGHTS_SAMPLING_PERCENTAGE || "100",
-  // The log level used for Winston logger (error, info, debug)
-  WINSTON_LOG_LEVEL: process.env.WINSTON_LOG_LEVEL || "debug",
-
-  // RESTful Webservice configuration
-  // These information are documented here:
-  // https://docs.google.com/document/d/1Qqe6mSfon-blHzc-ldeEHmzIkVaElKY5LtDnKiLbk80/edit
-  // Used to expose services
-  CONTROLLER: {
-    HOST: process.env.PAGOPAPROXY_HOST || localhost,
-    PORT: process.env.PAGOPAPROXY_PORT || process.env.PORT || 3000,
-    // SHA256 client certificate fingerprint (without `:` separators)
-    CLIENT_CERTIFICATE_FINGERPRINT:
-      process.env.PAGOPAPROXY_CLIENT_CERTIFICATE_FINGERPRINT,
-    ROUTES: {
-      RESTFUL: {
-        PAYMENT_ACTIVATIONS_GET:
-          "/payment-activations/:codiceContestoPagamento",
-        PAYMENT_ACTIVATIONS_POST: "/payment-activations",
-        PAYMENT_REQUESTS_GET: "/payment-requests/:rpt_id_from_string"
-      },
-      SOAP: {
-        PAYMENT_ACTIVATIONS_STATUS_UPDATE: "/FespCdService"
+const NODE_CONNECTION_CONFIG_PATH =
+  process.env.NODE_CONFIG_PATH || "./.node_config.json";
```

Comment:
it is not clear to me how this file can be generated starting from parameters in `keyvault`. maybe a step of pipeline that creates this file?

Reply:
p.s as an alternative, an environment variable with encoded json?

Reply:
I'd rather keep these parameters out of environment variables, for two reasons:
 * Editing JSON in .env (i.e. plain text) files is ugly 🙃 
 * Other configuration variables are more like _parameters_, while these influence _behaviour_ a bit, even though they all belong to configuration keeping them separated is not totally insane

As you pointed out this would require generating the file in the pipeline (or implementing a check for an env variable and falling back to a file, but I think this would be too brittle). With [jo](https://github.com/jpmens/jo) and a bit of bash is more than doable, but would make the pipeline more complex.

Alternatively we could just put everything into environment variables and call it a day

Reply:
yes, an environment variable with a json could be ugly  😁 But a file as part of the artifact with secrets is not a best practice, also remember: our pipelines are open [so](https://dev.azure.com/pagopaspa/pagoPA-projects/_build/results?buildId=18435&view=artifacts&pathAsName=false&type=publishedArtifacts) 🔥

Reply:
I was thinking something like this:
  - In the keyvault we have a secret that is a list of kv variables to use in the config
  - We [fetch the secrets in the keyvault](https://docs.microsoft.com/en-us/azure/devops/pipelines/release/key-vault-in-own-project?view=azure-devops&tabs=portal) before the Bash task that generates the file
  - We execute a bash task that uses the list of variables to collect them and pass them with their values to `jo`

This way everything would be still secret right?

Reply:
depends on the keyvault to use, Azure Devops and Azure App Service use two different kv, in your solution where would you save the secrets?

Reply:
I was thinking of the App Service KV

Reply:
unfortunately this is not possible for the configurations and policies that have been set in Azure Devops project, also I think that in this way we would distribute the app configuration on two separate kv

Reply:
Ok, I'll switch to env vars then